src/bencodeUtil.ts

```diff
@@ -0,0 +1,48 @@
+import * as bencoder from 'bencoder';
+
+const CONTINUATION_ERROR_MESSAGE: string = "Unexpected continuation: \"";
+
+interface DecodedResult {
+    decodedObjects: any[];
+    rest: Buffer;
+}
+
+export function encode(msg: any): Buffer {
+    return bencoder.encode(msg);
+}
+
+/*
+    receives a buffer and returns an array of decoded objects and the remaining unused buffer
+*/
+export function decodeObjects(buffer: Buffer): DecodedResult {
+    const decodedResult: DecodedResult = { decodedObjects: [], rest: buffer };
+    return decode(decodedResult);
+}
+
+function decode(decodedResult: DecodedResult): DecodedResult {
+    if (decodedResult.rest.length === 0)
+        return decodedResult;
+
+    try {
+        const decodedObj = bencoder.decode(decodedResult.rest);
+        decodedResult.decodedObjects.push(decodedObj);
+        decodedResult.rest = new Buffer('');
```

Comment:
@avli Good advice! I'm going to bump up vscode dev dependency version. The one we use is still on node 4.x which doesn't have those buffer improvements.